### PR TITLE
fix dotnet transitive outdated listing

### LINKED ISSUE #7890: "dotnet list package --outdated --include-transitive" also checks project references
When using "dotnet list package --outdated --include-transitive", project references are listed as "Not found at the sources", but it would make more sense to not list/check these at all.

Using dotnet.exe v2.2.201 on Windows 10.

### PR DESCRIPTION
**Pull request by zkat**
Fix https://github.com/NuGet/Home/issues/7890

## Bug

Fixes: https://github.com/NuGet/Home/issues/7890 
Regression: No   

## Fix

Details: This attempts to fix the issue by filtering out project ref packages from the table listing.

## Testing/Validation

Tests Added: Yes

### CODE REVIEW

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/ProjectPackagesPrintUtility.cs
Comment (nkolev92) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/ProjectPackagesPrintUtility.cs]: There are situations in which printing `Not found at sources` is valid. 
It's possible that the sources have been changed since the last restore.

Comment (zkat) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/ProjectPackagesPrintUtility.cs]: ah, so I should test specifically for package refs, then. I'll figure out how to do that :)

**Comment by zkat** [comment]
Update: progress on this is currently blocked by https://github.com/NuGet/Home/issues/8434

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/ProjectPackagesPrintUtility.cs
Comment (zkat) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/ProjectPackagesPrintUtility.cs]: Fixed by adding a Type field to the package objects and checking for that.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/InstalledPackageReference.cs
Comment (xavierdecoster) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/InstalledPackageReference.cs]: `Type` is a `string` property. The summary indicates this may have a value equal to `True`? Something's off :)

**Review by xavierdecoster — APPROVED** [review]

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/InstalledPackageReference.cs
Comment (zkat) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/InstalledPackageReference.cs]: hahaha, I had it as a bool at first. The docs, as always, lie. :) Thanks for the catch!

**Comment by zkat** [comment]
Note: this can't be merged until #3007 gets merged, and then it needs to be rebased.

**Comment by xavierdecoster** [comment]
@zkat think this one is unblocked ;)

**Comment by zkat** [comment]
Update: This is unblocked and ready for review

**Comment by nkolev92** [comment]
you might want to request reviews from the whole team :)

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/ProjectPackagesPrintUtility.cs
Comment (nkolev92) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/ProjectPackagesPrintUtility.cs]: Use LibraryType.Project
```suggestion
            packages = packages.Where(p => p.Name != LibraryType.Project);
```

That's what the original model LockFileTargetLibrary uses.

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs
Comment (nkolev92) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs]: I think the test name can be improved. 
Maybe something like: 

```suggestion
        public async Task DotnetListPackage_Outdated_ProjectReferences_AreNotDisplayed(string currentVersion, string args, string expectedVersion)
```

Comment (nkolev92) [reply test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs]: I have also created a tracking issue to add proper tests for dotnet list --deprecated.
https://github.com/NuGet/Home/issues/8481

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs
Comment (nkolev92) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs]: nit: remove args, it's not really used anywhere.

**Review by nkolev92 — COMMENTED** [review]
Looks good. 
Some suggestions before approving.

**Review by zivkan — APPROVED** [review]

**Review by nkolev92 — APPROVED** [review]

**Comment by zkat** [comment]
Note: holding off on merging until after 5.3!

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs
Comment (dtivel) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs]: Why not use `[PlatformFact(Platform.Windows)]`?

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs
Comment (dtivel) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs]: var versions = new List<string> { currentVersion, expectedVersion };
?

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs
Comment (dtivel) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs]: For all of these `True(…)`/`False(…)` assertions, how about using the `Assert` overload which takes a string message and pass the full dotnet output (e.g.:  `listResult.AllOutput` here) on failure?  This will facilitate on-sight failure diagnosis.

**Comment by nkolev92** [comment]
@zkat 🔔 

Please either address feedback/tests or close.

**Comment by nkolev92** [comment]
I'm gonna close this for now. 
Please re-open if/when we get back to it. 

It's long stale at this point :) 
We are losing *nothing* by closing the PR :)

**Pull request closed without merging**
